src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -167,14 +168,15 @@ master_create_empty_shard(PG_FUNCTION_ARGS)
 	shardId = GetNextShardId();
 
 	/* if enough live groups, add an extra candidate node as backup */
-	{
-		uint32 primaryNodeCount = ActivePrimaryNodeCount();
+	workerNodeList = ActivePrimaryNodeList(NoLock);
```

Comment:
afaict there's no need for a lock, because it should be okay for shard placements to ignore some newly added node. But it is important that `list_length(workerNodeList) == primaryNodeCount` so avoid having two separate calls to `ActivePrimaryNodeList`